packages/asset-uploader/package.json

```diff
@@ -29,7 +30,10 @@
     "typescript": "4.7.4",
     "zod": "^3.17.3"
   },
-  "main": "lib/index.js",
+  "exports": {
```

Comment:
Note that TypeScript won't understand `exports`. To make it understand them (while there is a chance that you don't want it to understand them right now) you'd have to use `moduleResolution: node16`. This in turn would **require** you to also add a `types` condition here - in that mode TS doesn't read from the top-level fields (`package.json#types`) if `exports` are present. Enabling this would also require you to use explicit extensions within `src`.

Reply:
Is it newer typescript requirement? Caz now everything works fine.

Reply:
Not exactly - it works right now because you don't use `moduleResolution: node16`. But because you don't use it TS doesn't actually look into `package.json#exports` at all. It sticks to reading from top-level keys **and** file system layout. If you'd switch to `moduleResolutions: node16` then it would stop working because you have no `types` condition (**unless** the requested file has a sibling `.d.ts` file - which might be the case here cause you build those artifacts using `tsc`, and even in `dev` TS writes to the same dir as `tsup`, this is kinda accidental though).

My main concern here is the `/server` entrypoint. It might work because there is a file called like this at the root of the module but this won't work with dual conditions etc (could be fixed with `pkg/server/package.json` though).

So, I'm not as much concerned here about how it works **right now** but I'm worried that the fact that it works is a little bit accidental and that it might require further tweaking in the future

Reply:
Alright, let's do this in separate PR